lib/helpers/ui.dart

```diff
@@ -32,75 +33,39 @@ AlertDialog buildMissedWordsDialog(GameBloc gb, BuildContext context) {
       ]);
 }
 
-class SBDrawer extends StatelessWidget {
-  const SBDrawer({
-    Key key,
-  }) : super(key: key);
-
-  Text _buildDrawerItem(String t, {double size}) {
-    return Text(
-      t,
-      style: TextStyle(fontSize: (size != null) ? size : 18),
-    );
-  }
-
-  @override
-  Widget build(BuildContext context) {
-    var gb = Provider.of(context).game;
-    return Drawer(
-      child: ListView(
-        children: <Widget>[
-          DrawerHeader(
-            child: _buildDrawerItem("Spelling Bee", size: 27),
-            decoration: BoxDecoration(
-              color: Colors.yellow,
-            ),
-          ),
-          ListTile(
-            title: _buildDrawerItem("New game"),
-            onTap: () {
-              Navigator.pop(context);
-              showDialog(
-                  context: context,
-                  builder: (context) => buildMissedWordsDialog(gb, context));
-            },
-          ),
-          ListTile(
-            title: _buildDrawerItem("Rules"),
-            onTap: () {
-              Navigator.pop(context);
-              Navigator.push(context,
-                  buildPageTransition(scaffold(Rules(), context, false)));
-              // showDialog(
-              //     context: context,
-              //     builder: (context) => Rules());
-            },
-          )
-        ],
-      ),
-    );
-  }
-}
-
-Widget scaffold(Widget w, BuildContext context, main) {
-  return Scaffold(
-    drawer: main ? SBDrawer() : null,
-    appBar: AppBar(
-      title: Text("Spelling Bee"),
-      actions: main
-          ? <Widget>[
-              IconButton(
-                icon: Icon(Icons.help),
-                onPressed: () {
-                  Navigator.push(context,
-                      buildPageTransition(scaffold(Rules(), context, false)));
-                },
-              ),
-            ]
-          : null,
-    ),
-    body: w,
-  );
+Widget scaffold(Widget w, BuildContext context, GameBloc gameBloc) {
```

Comment:
Should I use `Provider.of(context).game` instead of passing in `GameBloc`?

Reply:
Yeah, I think that's better. Actually, this can be done inside buildMissedWordsDialog and GameBloc doesn't have to be passed in there either. I should have done this when I was refactoring with blocs. Thank you!